pypdf/generic/_data_structures.py

```diff
@@ -81,6 +83,11 @@
 from ._fit import Fit
 from ._utils import read_hex_string_from_stream, read_string_from_stream
 
+if sys.version_info >= (3, 11):
+    from typing import Self
+else:
+    from typing_extensions import Self
```

Comment:
This will not work as we need at least `typing_extensions` for this (we currently use 3.7.4.3): https://github.com/python/typing_extensions/commit/0d10fec91740924bb48cc4435df093c12a1c2368

Reply:
When I used ArrayObject, I add warnings from mypy pushing into using Self. Any proposal ?

Reply:
The only way I see is to raise the required version of `typing_extensions` - providing a local backport does not really seem to be worth the effort and just complicates license compliance.

Reply:
I' confused : tying_extensions is requested >= 4.7.1

Reply:
Who requests `typing_extensions>=4.7.1`? We state https://github.com/py-pdf/pypdf/blob/f43268734a529d4098e6258bf346148fd24c54f0/pyproject.toml#L32 in our regular dependencies, thus we should rely on this.

Reply:
https://github.com/py-pdf/pypdf/blob/f43268734a529d4098e6258bf346148fd24c54f0/requirements/dev.txt#L94

Reply:
this is also consistent with the reported installation in pytest logs

Reply:
This is what we are testing against, not the minimum required version. The actual required versions are part of the `pyproject.toml` file and maybe the `dev.in` file. `dev.txt` is just some sort of package version snapshot generated at some point in time used to ensure that CI always uses the same package versions.

Reply:
no objection to align pyproject.toml to at least 3.10, or better to 4.7.1 ?

Reply:
I recommend to keep it minimal, thus use `typing_extensions >= 4.0.0`.

Reply:
by the way, just below dataclass for python release <3.7 is stated which is normally obsolete. OK to remove it?

Reply:
No, this should be done in a separate PR - there are other statements in the `pyproject.toml` which still reference Python 3.6 (`python_requires`, `cryptography`/`PyCryptodome`) and would need a similar cleanup.